### PR TITLE
PR #4: Extract job_matcher.sh module (Phase 2 - Boundaries)

### PR DESCRIPTION
**Pull request by ebanerjeeTT**
Extracts subjob matching logic into a reusable module as part of Phase 2 (Boundaries modularization).

**Changes:**
- Create `modules/boundaries/job_matcher.sh` with `match_subjob(job_name, subjob_name, workflow_name)` function
- Handles Unicode dash normalization (Pd category → ASCII '-'), case-insensitive matching
- Matching strategies: exact, workflow/subjob format, endswith, contains
- Add tests: `tests/modules/boundaries/job_matcher_test.sh` (13 tests)
- Update test workflow to run modules tests and trigger on `modules/**` changes

**Merge Criteria:** No changes to `find_boundaries.sh` yet (pure extraction per plan). All tests pass.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: .github/actions/auto-triage/auto_triage/modules/boundaries/job_matcher.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+if [ -n "${_JOB_MATCHER_LOADED:-}" ]; then
+    return 0
+fi
+_JOB_MATCHER_LOADED=1
+
+# Normalize strings for matching: Unicode dashes → ASCII '-', lowercase.
+# Character set must match find_boundaries.sh jq filter exactly:
+# U+2010 U+2011 U+2012 U+2013 U+2014 U+2015 U+2212 U+FE58 U+FE63 U+FF0D.
+# Single python3 invocation for all three strings to avoid process-per-call overhead.
```
Comment (ttmchiou) [inline .github/actions/auto-triage/auto_triage/modules/boundaries/job_matcher.sh]: link python3 version used here for documentation purposes

**Review by ttmchiou — APPROVED** [review]

File: .github/actions/auto-triage/auto_triage/modules/boundaries/job_matcher.sh (hunk 1 above)
Comment (ebanerjeeTT) [reply .github/actions/auto-triage/auto_triage/modules/boundaries/job_matcher.sh]: done
